fms/modules/attention.py

```diff
@@ -262,32 +262,35 @@ def __init__(
         assert torch.distributed.is_initialized()
 
         rank, world_size = distributed.rank_and_world(group)
-        assert (
-            nheads % world_size == 0
-        ), "The number of heads must be divisible by world size"
+        assert nheads % world_size == 0 and (
+            kvheads % world_size == 0 or world_size % kvheads == 0
+        ), "The number of heads must be divisible by world size or the world size must be a multiple of kv heads"
```

Comment:
the comment doesn't seem to correctly match the logic (and could be formatted better)

Reply:
specifically:

* the comment says "or" in a spot where code says "and"
* it can be a multiple of or divisible by

Reply:
Fixing this in next commit